File: longestCommonPrefix.py
class Solution(object):
    def matchPrefix(self, str1, str2):
        i = 0
        j = 0
        prefix = ''
        while i<len(str1) and j<len(str2):
            if str1[i]!=str2[j]:
                break
            prefix += str1[i]
            i += 1
            j += 1
        return prefix
                
    
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ''
        prefix = strs[0]
        for i in range(1, len(strs)):
            prefix = self.matchPrefix(prefix, strs[i])
        return prefix
            
        
                
        

File: java/longestCommonPrefix.java
class Solution {
    public String findPrefix(String str1, String str2) {
        int i = 0, j = 0;
        String prefix = "";
        
        while(i<str1.length() && j<str2.length()) {
            if(str1.charAt(i)!=str2.charAt(j))
                break;
            prefix += str1.charAt(i);
            i++;
            j++;
        }
        
        return prefix;
    }
    
    public String longestCommonPrefix(String[] strs) {
        if(strs.length==0)
            return "";
        
        String prefix = strs[0];
        
        for(int i=0; i<strs.length; i++)
            prefix = findPrefix(prefix, strs[i]);
        
        return prefix;
    }
}